unittests/savanna_cluster.hpp

```diff
@@ -106,10 +108,26 @@ namespace savanna_cluster {
          for (size_t i=0; i<num_chains_to_final; ++i)
             auto b = produce_block();
 
-         BOOST_REQUIRE_EQUAL(lib_block->block_num(), pt_block->block_num());
+         BOOST_REQUIRE_EQUAL(lib_num(), pt_block->block_num());
       }
 
-      uint32_t lib_num() const { return lib_block->block_num(); }
+      // updates producers (producer updates will be propagated to connected nodes), and
+      // wait until one of the new producers is pending.
+      // return the index of the pending new producer (we assume no duplicates in producer list)
+      // -----------------------------------------------------------------------------------
+      size_t set_producers(const std::vector<account_name>& producers) {
+         tester::set_producers(producers);
+         account_name pending;
+         while (1) {
```

Comment:
Would be nice to have some large limit on this so the test can fail with a useful message instead of timing out. I realize you just moved this code, but still would be nice.

Reply:
done.